src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -4570,4 +4597,168 @@ namespace winrt::TerminalApp::implementation
+        }
+
+        // You may think to yourself, this is a great place to increase the
+        // width of the TabView artificially, to make room for the new tab item.
+        // However, we'll never get a message that the tab left the tab view
+        // (without being dropped). So there's no good way to resize back down.
+    }
+
+    // Method Description:
+    // - Called on the TARGET of a tab drag/drop. We'll unpack the DataPackage
+    //   to find who the tab came from. We'll then ask the Monarch to ask the
+    //   sender to move that tab to us.
+    winrt::fire_and_forget TerminalPage::_onTabStripDrop(winrt::Windows::Foundation::IInspectable /*sender*/,
+                                                         winrt::Windows::UI::Xaml::DragEventArgs e)
+    {
+        // Get the PID and make sure it is the same as ours.
+        if (const auto& pidObj{ e.DataView().Properties().TryLookup(L"pid") })
+        {
+            const auto pidStr{ winrt::unbox_value<winrt::hstring>(pidObj) };
+            uint32_t pidNum;
+            if (Utils::StringToUint(pidStr.c_str(), pidNum))
+            {
+                if (pidNum != GetCurrentProcessId())
+                {
+                    // The PID doesn't match ours. We can't handle this drop.
+                    co_return;
+                }
+            }
+        }
+        else
+        {
+            // No PID? We can't handle this drop. Bail.
+            co_return;
+        }
+
+        const auto& windowIdObj{ e.DataView().Properties().TryLookup(L"windowId") };
+        if (windowIdObj == nullptr)
+        {
+            // No windowId? Bail.
+            co_return;
+        }
+        const auto windowId{ winrt::unbox_value<winrt::hstring>(windowIdObj) };
+
+        // Convert the windowId to a number
+        uint32_t src;
+        if (!Utils::StringToUint(windowId.c_str(), src))
+        {
+            // Failed to convert the windowId to a number. Bail.
+            co_return;
+        }
+
+        // Figure out where in the tab strip we're dropping this tab. Add that
+        // index to the request. This is largely taken from the WinUI sample
+        // app.
+
+        // We need to be on OUR UI thread to figure out where we dropped
+        auto weakThis{ get_weak() };
+        co_await wil::resume_foreground(Dispatcher());
+        if (const auto& page{ weakThis.get() })
+        {
+            // First we need to get the position in the List to drop to
+            auto index = -1;
+
+            // Determine which items in the list our pointer is between.
+            for (auto i = 0u; i < _tabView.TabItems().Size(); i++)
+            {
+                if (const auto& item{ _tabView.ContainerFromIndex(i).try_as<winrt::MUX::Controls::TabViewItem>() })
+                {
+                    const auto posX{ e.GetPosition(item).X };
+                    const auto itemWidth{ item.ActualWidth() };
+                    if (posX - itemWidth < 0)
```

Comment:
oh interesting. I think we have an off by one! This explains my weird tab 1 concerns from above.

Dropping ANYWHERE to the left of tab 3's right edge

```
                      | here    |
                      v         v
/-----------,---------,---------\
|  pwsh     | oof     | barbaz  |
```

causes us to insert here:

```
                      | here
                      v
/-----------,---------,---------\
|  pwsh     | oof     | barbaz  |
```